lib/register.js

```diff
@@ -0,0 +1,56 @@
+(function() {
+    var ts = require('./typescript');
+    var extensions = ['.ts', '.tsx'];
+
+    path = require('path');
+    fs = require('fs');
+
+    loadFile = function(module, filename) {
+        var source = fs.readFileSync(filename, 'utf-8');
+        var configFile = ts.findConfigFile(filename);
+        var opts = {
+            module: ts.ModuleKind.CommonJS,
+            target: ts.ScriptTarget.ES5
+        };
+        if (configFile) {
+            var configFileContents = ts.readConfigFile(configFile);
+            opts = configFileContents.config;
+            opts.files = [];
+            opts = ts.parseConfigFile(opts, null, process.cwd()).options;
+        }
+        var answer = ts.transpile(source, opts);
+        return module._compile(answer, filename);
+    };
+
+    if (require.extensions) {
```

Comment:
According to the Node docs, this is deprecated. (See https://nodejs.org/api/globals.html#globals_require_extensions ). While it does say it won't be removed as the module API is locked, it still feels odd to start adding new features based on deprecated functionality.